src/Disks/ObjectStorages/IMetadataStorage.h

```diff
@@ -143,6 +151,12 @@ class IMetadataTransaction : private boost::noncopyable
         return nullptr;
     }
 
+    virtual TruncateFileOperationOutcomePtr truncateFile(const std::string & /* path */, size_t /* size */)
+    {
+        throwNotImplemented();
+        return nullptr;
```

Comment:
```suggestion
        throwNotImplemented();
```
as this method is `[noreturn]`